src/averageAges.js

```diff
@@ -55,7 +75,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const neededPersons = onlyWithSon
+    ? people.filter(person => person.sex === 'm')
+    : people;
+
+  const ages = neededPersons.map((person) => {
+    const mother = person.mother;
+
+    const findMother = female => {
+      return mother === female.name;
+    };
+
+    const differenceAges = people.find(findMother)
```

Comment:
avoid creating variables/functions in iteration methods, otherwise they will create over and over again